### PR TITLE
fuzzel: support building with librsvg backend

### PR DESCRIPTION
**Pull request by jtrees**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fu/fuzzel/package.nix [hunk 1]
```diff
@@ -54,14 +55,15 @@ stdenv.mkDerivation (finalAttrs: {
...
+  ++ lib.optional pngSupport libpng
+  ++ lib.optional (svgBackend == "librsvg") librsvg;
 
   mesonBuildType = "release";
 
   mesonFlags = [
     "-Denable-cairo=${if enableCairo then "enabled" else "disabled"}"
     "-Dpng-backend=${if pngSupport then "libpng" else "none"}"
-    "-Dsvg-backend=${if svgSupport then "nanosvg" else "none"}"
+    "-Dsvg-backend=${svgBackend}"
```
Comment (getchoo) [inline pkgs/by-name/fu/fuzzel/package.nix:66]: Nitpick: we could also use `(lib.mesonFeature "svg-backend" svgBackend)` here (and with the other flags)

**Review by getchoo — APPROVED** [review]
LGTM. We may want to look into making this backwards compatible with the previous override if backporting to 24.11, though -- it could be done pretty easily  with something like `svgBackend ? if svgSupport then "nanosvg" else "none”`

File: pkgs/by-name/fu/fuzzel/package.nix (hunk 1 above)
Comment (jtrees) [reply pkgs/by-name/fu/fuzzel/package.nix]: > We may want to look into making this backwards compatible with the previous override

That's a good idea. I reworked the commit guided by your suggestions. I wasn't able to implement them exactly as you proposed but hope I was able to capture the intent.

File: pkgs/by-name/fu/fuzzel/package.nix [hunk 2]
```diff
@@ -13,14 +13,16 @@
...
+, svgSupport ? true
+, svgBackend ? "nanosvg" # alternative: "librsvg"
+  # Optional dependencies
 , cairo
 , libpng
+, librsvg
 }:
 
-assert svgSupport -> enableCairo;
+assert (svgBackend == "nanosvg") -> enableCairo;
```
Comment (getchoo) [inline pkgs/by-name/fu/fuzzel/package.nix:25]:
```suggestion
assert (svgSupport && svgBackend == "nanosvg") -> enableCairo;
```

This should be properly guarded, as if `svg-backend` is `"none"`, we don't actually need to enable Cairo

File: pkgs/by-name/fu/fuzzel/package.nix [hunk 3]
```diff
@@ -54,14 +56,15 @@ stdenv.mkDerivation (finalAttrs: {
     tllist
     fcft
   ] ++ lib.optional enableCairo cairo
-    ++ lib.optional pngSupport libpng;
+  ++ lib.optional pngSupport libpng
+  ++ lib.optional (svgBackend == "librsvg") librsvg;
```
Comment (getchoo) [inline pkgs/by-name/fu/fuzzel/package.nix:60]:
```suggestion
  ++ lib.optional (svgSupport && svgBackend == "librsvg") librsvg;
```
Similar to the above

File: pkgs/by-name/fu/fuzzel/package.nix [hunk 4]
```diff
@@ -54,14 +56,15 @@ stdenv.mkDerivation (finalAttrs: {
...
 
   mesonBuildType = "release";
 
-  mesonFlags = [
-    "-Denable-cairo=${if enableCairo then "enabled" else "disabled"}"
-    "-Dpng-backend=${if pngSupport then "libpng" else "none"}"
-    "-Dsvg-backend=${if svgSupport then "nanosvg" else "none"}"
+  mesonFlags = with lib; [
+    (mesonOption "enable-cairo" (if enableCairo then "enabled" else "disabled"))
```
Comment (getchoo) [inline pkgs/by-name/fu/fuzzel/package.nix:65]:
```suggestion
    (mesonEnable "enable-cairo" enableCairo)
```
https://noogle.dev/f/lib/mesonEnable can do this for us

File: pkgs/by-name/fu/fuzzel/package.nix [hunk 5]
```diff
@@ -54,14 +56,15 @@ stdenv.mkDerivation (finalAttrs: {
...
+  ++ lib.optional pngSupport libpng
+  ++ lib.optional (svgBackend == "librsvg") librsvg;
 
   mesonBuildType = "release";
 
-  mesonFlags = [
-    "-Denable-cairo=${if enableCairo then "enabled" else "disabled"}"
-    "-Dpng-backend=${if pngSupport then "libpng" else "none"}"
-    "-Dsvg-backend=${if svgSupport then "nanosvg" else "none"}"
+  mesonFlags = with lib; [
```
Comment (getchoo) [inline pkgs/by-name/fu/fuzzel/package.nix:64]:
```suggestion
  mesonFlags =  [
```

Nitpick: `with lib;` is generally [not recommended](https://nix.dev/guides/best-practices#with-scopes) when dealing with multiple namespaces like this

**Review by getchoo — CHANGES_REQUESTED** [review]
I actually like this solution a lot better. Just a few notes

File: pkgs/by-name/fu/fuzzel/package.nix (hunk 2 above)
Comment (jtrees) [reply pkgs/by-name/fu/fuzzel/package.nix]: Fixed

File: pkgs/by-name/fu/fuzzel/package.nix (hunk 3 above)
Comment (jtrees) [reply pkgs/by-name/fu/fuzzel/package.nix]: Fixed

File: pkgs/by-name/fu/fuzzel/package.nix (hunk 4 above)
Comment (jtrees) [reply pkgs/by-name/fu/fuzzel/package.nix]: Ok

File: pkgs/by-name/fu/fuzzel/package.nix (hunk 5 above)
Comment (jtrees) [reply pkgs/by-name/fu/fuzzel/package.nix]: Ok, I removed it again.

**Review by getchoo — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356621`

---
### `aarch64-linux`

**Pull request merged**